### ISSUE TITLE
MacOs compilation issues for Tools/rules.bzl 

### ISSUE THREAD

**Issue by criver** [cython/cython#2053]
Hey guys, 

Thanks a lot for defining bazel rules for `cython`. It is very nice. 

There is one issue though. When I compile cython in MacOS, the compiled `so` file has extension `cpython-36m-darwin.so` and break the script. 
The following code fixed the problem:
```
            + """ && python -c 'import os, sys, glob; n = len(sys.argv); find_so_file = lambda src: glob.glob("{}*so".format(src.split(".")[0]))[0]; [os.rename(find_so_file(src), dst) for src, dst in zip(sys.argv[1:], sys.argv[1+n//2:])]' $(SRCS) $(OUTS)""",
```

**Comment by scoder** [comment]
That extension is correct. What exactly is not working for you?

**Comment by criver** [comment]
The renaming command assumes no extension in the `so` file name.

**Comment by au-hao** [comment]
Currently, rules.bzl produces an error on renaming that looks like:

```
FileNotFoundError: [Errno 2] No such file or directory: 'truffle/utils_native.so' -> 'bazel-out/darwin-fastbuild/genfiles/truffle/utils_native.so'
```

This is because the rename script doesn't account for the extension being there. It expects the extension to be a raw `.so` instead of `.cpython-36m-darwin.so`

**Comment by Alfus** [comment]
I fixed this in my repo using glob and updating the rule command as such:
```
        cmd = "PYTHONHASHSEED=0 $(location @cython//:cythonize) -k -i %s $(SRCS)" % extra_flags +
              # Rename outputs to expected location.
              """ && python3 -c 'import os, sys, glob; n = len(sys.argv); [os.rename(glob.glob(src.split(".")[0] + ".*.so")[0], dst) for src, dst in zip(sys.argv[1:], sys.argv[1+n//2:])]' $(SRCS) $(OUTS)""",
```

**Comment by gudmundur-heimisson** [comment]
I am having the same issue on Fedora, the file generated has the extension `.cpython-311-x86_64-linux-gnu.so` rather than `.so`.

**Comment by scoder** [comment]
> the file generated has the extension `.cpython-311-x86_64-linux-gnu.so` rather than `.so`.

That is intended. Binary modules are platform specific in Python, so this file extension helps finding the right file to load for the current Python version and system.

**Comment by gudmundur-heimisson** [comment]
Yes, but the script breaks on any file that doesn’t end in a plain .so, so
the build fails.
